minimum_jerk_trajectory_planner/src/robot.cpp

```diff
@@ -0,0 +1,49 @@
+#include "minimum_jerk_trajectory_planner/robot.hpp"
+
+namespace minimum_jerk
+{
+
+    Robot::Robot(char *name, double total_time, TrajectoryPlanner &path_finder_controller, Pose& pose_start, Pose& pose_target, char *move_type) : name(name), path_finder_controller(path_finder_controller), total_time(total_time), pose(pose_start), pose_start(pose_start), pose_target(pose_target), move_type(move_type)
+    {
+        current_step = 0;
+        is_at_target = false;
+        odometry = nullptr;
```

Comment:
It's already nullptr since you never constructed it using `make_unique`, unnecessary ;)